llvm/test/CodeGen/WebAssembly/tailcall.ll

```diff
@@ -296,12 +296,13 @@ define i32 @mismatched_byval(ptr %x) {
 ; CHECK-NEXT:    global.set __stack_pointer, $pop7
 ; CHECK-NEXT:    i32.load $push0=, 0($0)
 ; CHECK-NEXT:    i32.store 12($1), $pop0
+; CHECK-NEXT:    i32.const $push5=, 12
+; CHECK-NEXT:    i32.add $push6=, $1, $pop5
+; CHECK-NEXT:    call $0=, quux, $pop6
```

Comment:
It seems that if this were actually a "matched" byval (i.e. the caller had a byval param that was passed directly to the callee's byval param) then it could work, since the caller wouldn't need to allocate space on its own stack for the byval.
I'm not sure it would be worth implementing that though.